### DISCUSSION TITLE
check permission in custom endpoints

### DISCUSSION THREAD

**Discussion by ldarren** [directus/directus#5830] [Archive]
what I intend to do in a custom endpoint
1. check permission for the requester to make order (throw error if no permission)
2. forward the request to external service to make the order
3. process response of external service (convert to order format)
4. save the order to database by using ItemsService

what I'm not sure is step `1`. how do I check permission of the requester?

I have tried the following ways:
1. inspect the `accountability` but I have the role only
2. used `PermissionsService.getAllowedFields`
```js
        const  service = new PermissionsService({ schema: req.schema, accountability: req.accountability });
        const pk = service.getAllowedFields('create', 'order')
       console.log(pk) // pk is always an empty object
```
is `step 1` achievable currently? a code sample will be very helpful to me, Thanks in advance

**Comment by rijkvanzanten** [comment]
The current user's permissions can be found under schema 🙂 

Something like this should do:

```js
let hasPermission = false;

for (const permission of req.schema.permissions) {
  if (permission.collection === 'order' && permission.action === 'create') {
    hasPermission = true;
    break;
  }
}
```
